driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -411,14 +418,16 @@ private boolean mechanismRequiresPassword(@Nullable final AuthenticationMechanis
      * @param <T>                    the mechanism property type
      */
     <T> MongoCredential(final MongoCredential from, final String mechanismPropertyKey, final T mechanismPropertyValue) {
-        notNull("mechanismPropertyKey", mechanismPropertyKey);
-
-        this.mechanism = from.mechanism;
-        this.userName = from.userName;
-        this.source = from.source;
-        this.password = from.password;
-        this.mechanismProperties = new HashMap<>(from.mechanismProperties);
-        this.mechanismProperties.put(mechanismPropertyKey.toLowerCase(), mechanismPropertyValue);
+        this(from.mechanism, from.userName, from.source, from.password, mapWith(
```

Comment:
Are there existing regression tests for this constructor? It doesn't look the semantics have changed, but would be good to have a test that asserts it.

Reply:
Yes, I believe so. This constructor is used only in:
```
    public <T> MongoCredential withMechanismProperty(final String key, final T value) {
        return new MongoCredential(this, key, value);
    }
```
which has 43 usages, over half of which are directly in tests, including `MongoCredentialSpecification`. I broke it by commending out `result.put(key, value);`, which caused existing tests to fail.